removed extra EnableAnalyzers props

vlada-shubina opened the pull request:
fixes https://github.com/dotnet/sdk/issues/27735

vlada-shubina committed:
removed extra EnableAnalyzers props

GangWang01 commented on src/Cli/Microsoft.TemplateEngine.Cli/Microsoft.TemplateEngine.Cli.csproj:
```diff
@@ -4,7 +4,6 @@
     <Description>Template creation for the dotnet CLI</Description>
     <IsPackable>true</IsPackable>
     <IsShippingPackage>false</IsShippingPackage>
-    <EnableAnalyzers>true</EnableAnalyzers>
```
It is not extra. It enables analyzers specified in `eng/Analyzers.props` for build in development. For sourcebuild, it needs to overwrite the value when `DotNetBuildFromSource` is true in template engine projects.
```
<EnableAnalyzers>true</EnableAnalyzers>
<EnableAnalyzers Condition="'$(DotNetBuildFromSource)' == 'true'">false</EnableAnalyzers>
```

vlada-shubina committed:
fixed analyzer enablement

vlada-shubina commented:
okay, I fixed it and enabled `Microsoft.CodeAnalysis.CSharp.CodeStyle` solution wide.
There were only few violations and I fixed them.
codeowners, @dotnet/dotnet-cli  please shout if you are not okay with this change,

vlada-shubina commented:
I decided to move to `<EnforceCodeStyleInBuild>` property instead adding the reference - it looks simpler.

vlada-shubina commented:
`CodeStyle` analyzer started to have problems in VS. Might be related to https://github.com/dotnet/sdk/pull/27747.

@MichaelSimons please let me know if this is urgent. If so, I can do another PR fixing StyleCop only and removing `CodeStyle` until the issue is fixed.

MichaelSimons commented:
> @MichaelSimons please let me know if this is urgent. If so, I can do another PR fixing StyleCop only and removing `CodeStyle` until the issue is fixed.

A fix is needed for source-build by RC2.  We can't ship RC2 with prebuilts.

vlada-shubina commented:
https://github.com/dotnet/sdk/pull/27747 is merged, so merging this one too.

ViktorHofer commented on src/ApiCompat/Microsoft.DotNet.ApiCompatibility/IAssemblySymbolLoader.cs:
```diff
@@ -1,7 +1,9 @@
 ﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+#if NET472_OR_GREATER
```
@vlada-shubina curious, why was this change needed?

vlada-shubina replied:
good catch. This directive can be removed. 
Strangely enough it is not shown as an error in full framework, only in .NET.

vlada-shubina replied:
I'll fix it in another PR shortly.

vlada-shubina replied:
Fixed in https://github.com/dotnet/sdk/pull/27795/commits/1b030c7591e7f866530fcc1f11cc8dce71d98387
